apis/infrastructure/v1beta1/bootstrapkubeconfig_types.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// DefaultClusterName is the cluster name used in the generated bootstrap-kubeconfig
+	DefaultClusterName = "default-cluster"
+
+	// DefaultContext is the context name used in the generated bootstrap-kubeconfig
+	DefaultContext = "default-context"
+
+	// DefaultNamespace is the namespace used in the generated bootstrap-kubeconfig
+	DefaultNamespace = "default"
+
+	// DefaultAuth is the auth in the generated bootstrap-kubeconfig
+	DefaultAuth = "default-auth"
+
+	// BootstrapTokenDescription is the bootstrap token description for the secret that is generated
+	BootstrapTokenDescription = "Token generated by cluster-api-provider-bringyourownhost"
+
+	// BootstrapTokenExtraGroups is the byoh group that has access to create CertificateSigningRequest
+	BootstrapTokenExtraGroups = "system:bootstrappers:byoh"
+)
+
+// BootstrapKubeconfigSpec defines the desired state of BootstrapKubeconfig
+type BootstrapKubeconfigSpec struct {
+	// Server is the address of the kubernetes cluster (https://hostname:port).
+	Server string `json:"server"`
+
+	// InsecureSkipTLSVerify skips the validity check for the server's certificate. This will make your HTTPS connections insecure.
+	// +optional
+	// +kubebuilder:default=false
+	InsecureSkipTLSVerify bool `json:"insecure-skip-tls-verify,omitempty"`
+
+	// CertificateAuthorityData contains PEM-encoded certificate authority certificates.
+	CertificateAuthorityData string `json:"certificate-authority-data,omitempty"`
+}
+
+// BootstrapKubeconfigStatus defines the observed state of BootstrapKubeconfig
+type BootstrapKubeconfigStatus struct {
```

Comment:
It would be worth following API conventions and declaring field as pointers if all the fields are optional, as mentioned in https://cluster-api.sigs.k8s.io/contributing#api-conventions.
We can follow this for new APIs added and maybe taken care separately for existing APIs.